File: app/src/main/java/com/extremedesign/typeracer/DataRepository/RoomDatabase/TypeRacerDatabase.java
package com.extremedesign.typeracer.DataRepository.RoomDatabase;

import android.content.Context;

import androidx.annotation.NonNull;
import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;
import androidx.sqlite.db.SupportSQLiteDatabase;

import com.extremedesign.typeracer.model.ProfileImage;

@Database(entities = {ProfileImage.class},version = 1)
public abstract class TypeRacerDatabase extends RoomDatabase {
    private static TypeRacerDatabase INSTANCE;
    public abstract TypeRacerDAO typeRacerDAO();

    public static TypeRacerDatabase getDatabase(Context context){
        if(INSTANCE==null){
            synchronized (TypeRacerDatabase.class){
                if(INSTANCE==null){
                    INSTANCE= Room.databaseBuilder(context, TypeRacerDatabase.class,"questions_db")
                            .addCallback(new Callback() {
                                @Override
                                public void onOpen(@NonNull SupportSQLiteDatabase db) {
                                    super.onOpen(db);
                                }
                            })
                            .fallbackToDestructiveMigration()
                            .build();
                }
            }

        }

        return INSTANCE;
    }

}

File: app/src/main/java/com/extremedesign/typeracer/data_repository/room_database/TypeRacerDatabase.kt
package com.extremedesign.typeracer.data_repository.room_database

import android.content.Context
import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase
import com.extremedesign.typeracer.model.ProfileImage
import com.extremedesign.typeracer.model.User

@Database(entities = [ProfileImage::class, User::class], version = 1)
abstract class TypeRacerDatabase : RoomDatabase() {
    abstract fun typeRacerDAO(): TypeRacerDAO

    companion object {
        private var INSTANCE: TypeRacerDatabase? = null

        fun getDatabase(context: Context?): TypeRacerDatabase {
            if (INSTANCE == null) {
                synchronized(TypeRacerDatabase::class.java) {
                    if (INSTANCE == null) {
                        INSTANCE = Room.databaseBuilder(context!!, TypeRacerDatabase::class.java, "typeracer_db")
                                .addCallback(object : RoomDatabase.Callback() {})
                                .fallbackToDestructiveMigration()
                                .build()
                    }
                }
            }
            return INSTANCE!!
        }
    }
}